src/components/components/CommonComponents.js

```diff
@@ -21,6 +22,14 @@ export class CommonComponents extends React.Component {
     entity: React.PropTypes.object
   };
 
+  componentDidMount () {
+    Events.on('selectedEntityComponentChanged', detail => {
+      if (DEFAULT_COMPONENTS.indexOf(detail.name) !== -1) {
+        this.forceUpdate();
```

Comment:
@ngokevin changed! It seems to works perfect, as you change the state it will trigger render again.